internal/group/forum/httphandler.go

```diff
@@ -0,0 +1,139 @@
+package forum
+
+import (
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/jwtauth/v5"
+
+	"net/http"
+
+	"github.com/go-chi/render"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/np-inprove/server/internal/apperror"
+	"github.com/np-inprove/server/internal/config"
+	"github.com/np-inprove/server/internal/middleware"
+	"github.com/np-inprove/server/internal/payload"
+)
+
+type httpHandler struct {
+	service UseCase
+	cfg     *config.Config
+	jwt     *jwtauth.JWTAuth
+}
+
+func NewHTTPHandler(u UseCase, c *config.Config, j *jwtauth.JWTAuth) chi.Router {
+	a := &httpHandler{u, c, j}
+	r := chi.NewRouter()
+
+	// Authenticated routes
+	r.Group(func(r chi.Router) {
+		r.Use(jwtauth.Verify(j, func(r *http.Request) string {
+			c, err := r.Cookie(c.AppJWTCookieName())
+			if err != nil {
+				return ""
+			}
+			return c.Value
+		}))
+
+		r.Use(middleware.Authenticator)
+
+		r.Get("/", a.ListForums)
+		r.Post("/", a.CreateForum)
+		r.Delete("/{shortName}", a.DeleteForum)
+		r.Put("/{shortName}", a.UpdateForum)
+	})
+
+	return r
+}
+
+func (h httpHandler) ListForums(w http.ResponseWriter, r *http.Request) {
+	token := r.Context().Value(jwtauth.TokenCtxKey)
+	email := token.(jwt.Token).Subject()
+
+	forums, err := h.service.ListPrincipalForums(r.Context(), email)
```

Comment:
@tan-yun-e This would be something like,

```suggestion
	shortName = chi.URLParam(r, "shortName")
	forums, err := h.service.ListForums(r.Context(), email, shortName)
```

Reply:
That command is used to fetch the shortname for the forum in UpdateForum, how would it look like for UpdateForum?

Reply:
..? Not sure what you mean by UpdateForum

Reply:
The signature for all httphandler functions are the same 

w http.ResponseWriter
r *http.Request